docs/developing/integrate/wallets/integrating_walletconnect.mdx

```diff
@@ -63,84 +65,81 @@ For this guide, we are configuring a React App.
 
-```javascript
-import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
-import { configureChains, createClient, WagmiConfig } from 'wagmi'
-import { neonDevnet } from '@wagmi/chains'
-
-const chains = [neonDevnet]
-const projectId = "<import_from_env>"
-
-const { provider } = configureChains(chains, [w3mProvider({ projectId })])
-const wagmiClient = createClient({
-    autoConnect: true,
-    connectors: w3mConnectors({ projectId, version: 1, chains }),
-    provider
-})
-```
+3.4 Create `index.html` file and place the following code:
 
-3.4 Pass your ethereumClient and `projectId` to the Web3Modal component:
+```html
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <title>Neon WalletConnect Example</title>
+    </head>
+    <body>
+        <w3m-button />
+        <script type="module" src="main.js"></script>
+    </body>
+</html>
+```
 
-<Tabs>
-<TabItem value="React/Next" label="app.jsx" default>
+3.5 Create `main.js` file and place the following code:
 
 ```javascript
-import { Web3Modal } from '@web3modal/react'
-
-const web3Modal = new Web3Modal({ projectId }, ethereumClient); // for using in code
-const ethereumClient = new EthereumClient(wagmiClient, chains)
-
-function App() {
-  return <>
-    <WagmiConfig client={wagmiClient}>
-      <HomePage />
-    </WagmiConfig>
-    <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
-  </>
-}
-```
+import { createWeb3Modal, defaultWagmiConfig } from '@web3modal/wagmi';
+import { neonMainnet, neonDevnet } from '@wagmi/core/chains';
 
-</TabItem>
-<TabItem value="HTML" label="HTML">
+const projectId = import.meta.env.VITE_PROJECT_ID;
+const metadata = {
+    name: 'Neon Example',
+    description: 'Neon Web3Modal Example',
+    url: 'https://neonevm.org',
+    icons: ['https://neonevm.org/favicons/android-chrome-512x512.png']
+};
 
-```html
-<body>
-  <div id="app">
-    <w3m-core-button></w3m-core-button>
-  </div>
-  <script type="module" src="main.js"></script>
-</body>
- ```
+const chains = [neonMainnet, neonDevnet];
+const wagmiConfig = defaultWagmiConfig({chains, projectId, metadata});
 
-</TabItem>
-</Tabs>
+const modal = createWeb3Modal({wagmiConfig, projectId, chains});
+```
 
+3.6 Run command `npm run dev` to start your Web application and you should see the Web3Modal button loaded in your Web app page.
 
 ### Step 4: Connect your wallet
+For this example we will be using [MathWallet](https://mathwallet.org/).
 
-Start your application: `npm/yarn start`
+4.1 After clicking the Web3Modal button you should be able to see several connecting options, select WalletConnect:
+<img src={buttonclicked} width="300" />
 
-You may now  connect your preferred wallet with Neon EVM Devnet within the WalletConnect interface.
+4.2 After selecting WalletConnect the popup will load a WalletConnect QR code:
+<img src={walletconnectqr} width="300" />
```

Comment:
Can you cut your images closer to remove the grey background (or post-process it gone)?

Reply:
@mnedelchev-vn PTAL at this request

Reply:
done